controllers/flinkcluster_util.go

```diff
@@ -129,6 +171,70 @@ func getFromSavepoint(jobSpec batchv1.JobSpec) string {
 	return ""
 }
 
+// newRevision generates FlinkClusterSpec patch and makes new child ControllerRevision resource with it.
+func newRevision(cluster *v1beta1.FlinkCluster, revision int64, collisionCount *int32) (*appsv1.ControllerRevision, error) {
```

Comment:
Can we add unit tests for these util functions?

Reply:
Done.